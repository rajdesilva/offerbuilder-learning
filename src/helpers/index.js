export * from './api-config';
export * from './history';
