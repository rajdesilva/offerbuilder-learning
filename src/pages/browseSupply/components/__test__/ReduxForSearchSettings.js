export const ReduxForSearchSettings = {
  searchparams: {
    destination: {},
    los: 1,
    distance: 10,
    onlySupplier: false,
    target: {
      suppliers: [],
      channels: [],
    },
    brands: [
      {
        storefronts: [{}],
      },
    ],
    dateRange: {
      startDate: "05/17/2021",
      endDate: "07/17/2021",
    },

    lcn: false,
    remainingCapitalPool: "",
    sortBy: {
      sortOrder: "ASCENDING",
      sortCriteria: "NAME",
    },
  },
  channelinfo: {
    loading: false,
    channels: [
      {
        id: "PCT",
        name: "PCT",
      },
      {
        id: "PCT",
        name: "ATG",
      },
      {
        id: "DEMO_OFFERBUILDER",
        name: "DEMO_OFFERBUILDER",
      },
    ],
  },
  sourceinfo: {
    loading: false,
    brands: [
      {
        id: "idb",
        name: "Internal Demo Brand",
        storefronts: [
          {
            id: "ids",
            name: "Internal Demo Storefront",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids2",
            name: "Best Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids3",
            name: "PC Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
        ],
      },
      {
        id: "cdb",
        name: "Client Demo Brand",
        storefronts: [
          {
            id: "cds",
            name: "Client Demo Storefront",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids2",
            name: "Best Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids3",
            name: "PC Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
  },
};
