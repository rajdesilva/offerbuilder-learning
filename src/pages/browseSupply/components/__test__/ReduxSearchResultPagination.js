export const PaginationStore = {
  searchedproperties: {
    loading: false,
    properties: [{}],
    totalMatch: 15,
  },
  searchparams: {
    pageOffset: 0,
    pageSize: 10,
  },
};
