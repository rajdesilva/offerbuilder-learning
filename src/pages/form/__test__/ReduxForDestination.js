export const reduxStore = {
  supplysearchParams: {
    destination: {},
    los: 1,
    onlySupplier: true,
    target: {
      suppliers: [],
      channels: [],
    },
    dateRange: {
      startDate: "2020-09-18",
      endDate: "2020-11-24",
    },

    lcn: false,
    remainingCapitalPool: "",
  },
  channels: {
    // loading: false,
    channels: [
      {
        id: "AT_CORDIAL",
        name: null,
        type: "Portal",
        status: "HIDDEN",
      },
      {
        id: "AT_KINDERHOTELS",
        name: "Ihr Bett im Allg√§u",
        type: "Portal",
        status: "HIDDEN",
      },
    ],
  },
};
