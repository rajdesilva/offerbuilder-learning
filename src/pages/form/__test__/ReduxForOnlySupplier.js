export const ReduxForOnlySupplier = {
  searchparams: {
    destination: {},
    los: 1,
    onlySupplier: true,
    target: {
      suppliers: [
        {
          id: "ntp",
          name: "NTP",
        },
      ],
      channels: [
        {
          id: "AT_CORDIAL",
          name: "",
        },
      ],
    },
    dateRange: {
      startDate: "2020-09-18",
      endDate: "2020-11-24",
    },

    lcn: false,
    remainingCapitalPool: "",
  },
  channelinfo: {
    loading: false,
    channels: [
      {
        id: "AT_CORDIAL",
        name: "",
        type: "Portal",
        status: "HIDDEN",
      },
      {
        id: "india_channel",
        name: "indian channel",
        type: "Portal",
        status: "HIDDEN",
      },
    ],
  },
};
