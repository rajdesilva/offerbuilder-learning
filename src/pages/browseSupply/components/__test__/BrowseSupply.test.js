import React from "react";
import { createStore } from "redux";
import { render } from "../../../../helpers/testUtils";
import { MemoryRouter } from "react-router-dom";
import { BrowseSupply } from "../BrowseSupply";
import { tempStore } from "../../../offers/components/__test__/tempStore";

window.fetchWrapper = window.fetchWrapper = jest.fn().mockResolvedValue({
  success: true,
});

describe("BrowseSupply component test", () => {
  test("create BrowseSupply components snapshot", () => {
    const store = createStore(() => ({ ...tempStore }));
    const wrapper = render(
      <MemoryRouter initialEntries={["/browse-supply/"]}>
        <BrowseSupply />
      </MemoryRouter>,
      {
        store,
      }
    );
    const { container } = wrapper;
    expect(container.firstChild).toMatchSnapshot();
  });

  test("create BrowseSupply - search-result - verify text on the components", () => {
    const store = createStore(() => ({ ...tempStore }));
    const wrapper = render(
      <MemoryRouter initialEntries={["/browse-supply/search-results"]}>
        <BrowseSupply />
      </MemoryRouter>,
      {
        store,
      }
    );

    const { getByText } = wrapper;
    expect(getByText("Browse Supply")).toBeInTheDocument();
    expect(getByText("Search Parameters")).toBeInTheDocument();
    expect(getByText(/Lowest Price On/i)).toBeInTheDocument();
  });
});
