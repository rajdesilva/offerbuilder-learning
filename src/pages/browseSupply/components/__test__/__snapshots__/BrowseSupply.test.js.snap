// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowseSupply component test create BrowseSupply components snapshot 1`] = `
<div
  data-testid="browse-supply"
>
  <div
    class="ant-row ant-row-space-between"
  >
    <div
      class="ant-col"
    >
      <h2>
        Browse Supply
      </h2>
    </div>
  </div>
  <form
    class="ant-form ant-form-horizontal custom-form"
    data-testid="search-setting-form"
  >
    <div
      class="ant-row"
      style="margin: -4px -4px 4px -4px;"
    >
      <div
        class="ant-col ant-col-4 col1-text"
        style="padding: 4px 4px 4px 4px;"
      >
        Destination
      </div>
      <div
        class="ant-col ant-col-8"
        style="padding: 4px 4px 4px 4px;"
      >
        <div
          class="ant-row ant-form-item"
          data-testid="destination"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-select ant-select-auto-complete ant-select-single ant-select-customize-input ant-select-show-search"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <span
                        class="ant-input-affix-wrapper ant-select-selection-search-input"
                      >
                        <input
                          aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                          aria-autocomplete="list"
                          aria-controls="rc_select_TEST_OR_SSR_list"
                          aria-haspopup="listbox"
                          aria-owns="rc_select_TEST_OR_SSR_list"
                          autocomplete="off"
                          class="ant-input"
                          data-testid="search-city-input"
                          id="rc_select_TEST_OR_SSR"
                          placeholder="Search City"
                          role="combobox"
                          type="search"
                          value=""
                        />
                        <span
                          class="ant-input-suffix"
                        >
                          <span
                            aria-label="close-circle"
                            class="anticon anticon-close-circle ant-input-clear-icon-hidden ant-input-clear-icon"
                            role="button"
                            tabindex="-1"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="close-circle"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                              />
                            </svg>
                          </span>
                        </span>
                      </span>
                    </span>
                    <span
                      class="ant-select-selection-placeholder"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-col ant-col-4"
        style="padding: 4px 4px 4px 4px;"
      >
        <div
          class="ant-row ant-form-item"
          data-testid="distance"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-select ant-select-single ant-select-show-arrow"
                  data-testid="test-distance"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                        aria-autocomplete="list"
                        aria-controls="rc_select_TEST_OR_SSR_list"
                        aria-haspopup="listbox"
                        aria-owns="rc_select_TEST_OR_SSR_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="rc_select_TEST_OR_SSR"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-item"
                      title="10 km"
                    >
                      10 km
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
                 
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row"
      style="margin: -4px -4px 4px -4px;"
    >
      <div
        class="ant-col ant-col-4 col1-text"
        style="padding: 4px 4px 4px 4px;"
      >
        Hotel Name
      </div>
      <div
        class="ant-col ant-col-8"
        style="padding: 4px 4px 4px 4px;"
      >
        <div
          class="ant-row ant-form-item"
          data-testid="hotel-name"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input"
                  data-testid="hotel-name"
                  maxlength="200"
                  placeholder="Enter name"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row"
      style="margin: -4px -4px 4px -4px;"
    >
      <div
        class="ant-col ant-col-4 col1-text"
        style="padding: 4px 4px 4px 4px;"
      >
        Date Range
      </div>
      <div
        class="ant-col ant-col-8"
        style="padding: 4px 4px 4px 4px;"
      >
        <div
          class="ant-picker ant-picker-range"
          data-testid="range-picker"
        >
          <div
            class="ant-picker-input ant-picker-input-active"
          >
            <input
              autocomplete="off"
              placeholder="Start date"
              readonly=""
              size="12"
              value="04/07/2021"
            />
          </div>
          <div
            class="ant-picker-range-separator"
          >
            <span
              aria-label="to"
              class="ant-picker-separator"
            >
              <span
                aria-label="swap-right"
                class="anticon anticon-swap-right"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="swap-right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="0 0 1024 1024"
                  width="1em"
                >
                  <path
                    d="M873.1 596.2l-164-208A32 32 0 00684 376h-64.8c-6.7 0-10.4 7.7-6.3 13l144.3 183H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h695.9c26.8 0 41.7-30.8 25.2-51.8z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <div
            class="ant-picker-input"
          >
            <input
              autocomplete="off"
              placeholder="End date"
              readonly=""
              size="12"
              value="06/06/2021"
            />
          </div>
          <div
            class="ant-picker-active-bar"
            style="left: 0px; width: 0px; position: absolute;"
          />
          <span
            class="ant-picker-suffix"
          >
            <span
              aria-label="calendar"
              class="anticon anticon-calendar"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="calendar"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                />
              </svg>
            </span>
          </span>
          <span
            class="ant-picker-clear"
          >
            <span
              aria-label="close-circle"
              class="anticon anticon-close-circle"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="close-circle"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                />
              </svg>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      class="ant-row"
      style="margin: -4px -4px 4px -4px;"
    >
      <div
        class="ant-col ant-col-4 col1-text"
        style="padding: 4px 4px 4px 4px;"
      >
        Length Of Stay
      </div>
      <div
        class="ant-col ant-col-4"
        style="padding: 4px 4px 4px 4px;"
      >
        <div
          class="ant-input-number"
        >
          <div
            class="ant-input-number-handler-wrap"
          >
            <span
              aria-label="Increase Value"
              class="ant-input-number-handler ant-input-number-handler-up"
              role="button"
              unselectable="on"
            >
              <span
                aria-label="up"
                class="anticon anticon-up ant-input-number-handler-up-inner"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="up"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                  />
                </svg>
              </span>
            </span>
            <span
              aria-disabled="true"
              aria-label="Decrease Value"
              class="ant-input-number-handler ant-input-number-handler-down ant-input-number-handler-down-disabled"
              role="button"
              unselectable="on"
            >
              <span
                aria-label="down"
                class="anticon anticon-down ant-input-number-handler-down-inner"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <div
            class="ant-input-number-input-wrap"
          >
            <input
              aria-valuemax="1000"
              aria-valuemin="1"
              aria-valuenow="1"
              autocomplete="off"
              class="ant-input-number-input"
              data-testid="length-of-stay"
              max="1000"
              min="1"
              placeholder="None"
              role="spinbutton"
              step="1"
              value="1"
            />
          </div>
        </div>
        <span>
            
          Nights
        </span>
      </div>
      <div
        class="ant-col ant-col-4"
        style="padding: 4px 4px 4px 4px;"
      />
    </div>
    <div
      class="ant-divider ant-divider-horizontal"
      role="separator"
    />
    <div
      class="ant-row"
      style="margin: -4px -4px 4px -4px;"
    >
      <div
        class="ant-col ant-col-4 col1-text"
        style="padding: 4px 4px 4px 4px;"
      >
        Target
      </div>
      <div
        class="ant-col ant-col-8"
        style="padding: 4px 4px 4px 4px;"
      >
        <button
          aria-checked="false"
          class="ant-switch"
          data-testid="supplier-switch"
          id="onlySupplier"
          role="switch"
          type="button"
        >
          <div
            class="ant-switch-handle"
          />
          <span
            class="ant-switch-inner"
          />
        </button>
        <span>
            
          Only Supplier
        </span>
      </div>
    </div>
    <div
      class="ant-row"
      style="margin: -4px -4px 4px -4px;"
    >
      <div
        class="ant-col ant-col-4 col1-text"
        style="padding: 4px 4px 4px 4px;"
      />
      <div
        class="ant-col ant-col-6"
        style="padding: 4px 4px 4px 4px;"
      >
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label"
          >
            <label
              class="ant-form-item-required"
              for="target_suppliers"
              title=""
            >
              Supplier
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-select ant-select-multiple ant-select-show-search"
                  data-testid="nemo-select-brw-search-supply"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                      style="width: 0px;"
                    >
                      <input
                        aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                        aria-autocomplete="list"
                        aria-controls="rc_select_TEST_OR_SSR_list"
                        aria-haspopup="listbox"
                        aria-owns="rc_select_TEST_OR_SSR_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="rc_select_TEST_OR_SSR"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                      <span
                        aria-hidden="true"
                        class="ant-select-selection-search-mirror"
                      >
                        
                         
                      </span>
                    </span>
                    <span
                      class="ant-select-selection-placeholder"
                    >
                      Please Select
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-col ant-col-7"
        style="padding: 4px 4px 4px 4px;"
      >
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label"
          >
            <label
              class="ant-form-item-required"
              for="target_channels"
              title=""
            >
              Channels Only For NTP
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-select ant-select-multiple ant-select-show-search"
                  data-testid="nemo-select-brw-search-channel"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                      style="width: 0px;"
                    >
                      <input
                        aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                        aria-autocomplete="list"
                        aria-controls="rc_select_TEST_OR_SSR_list"
                        aria-haspopup="listbox"
                        aria-owns="rc_select_TEST_OR_SSR_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="rc_select_TEST_OR_SSR"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                      <span
                        aria-hidden="true"
                        class="ant-select-selection-search-mirror"
                      >
                        
                         
                      </span>
                    </span>
                    <span
                      class="ant-select-selection-placeholder"
                    >
                      Please Select
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="ant-row"
        style="margin: -4px -4px 4px -4px;"
      >
        <div
          class="ant-col ant-col-4 col1-text"
          style="padding: 4px 4px 4px 4px;"
        />
        No clients found
      </div>
      <div
        class="ant-row"
        style="margin: -4px -4px 4px -4px;"
      >
        <div
          class="ant-col ant-col-6 col1-text"
          style="padding: 4px 4px 4px 4px;"
        />
        <div
          class="ant-col ant-col-4"
          style="padding: 4px 4px 4px 4px;"
        >
          <button
            class="ant-btn ant-btn-link"
            data-testid="add-brand"
            type="button"
          >
            <span
              aria-label="plus"
              class="anticon anticon-plus"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="plus"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <defs>
                  <style />
                </defs>
                <path
                  d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                />
                <path
                  d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                />
              </svg>
            </span>
            <span>
               
            </span>
            Add client
          </button>
        </div>
      </div>
    </div>
    <div
      class="ant-divider ant-divider-horizontal"
      role="separator"
    />
    <div
      class="ant-row"
      style="margin: -4px -4px 4px -4px;"
    >
      <div
        class="ant-col ant-col-4 col1-text"
        style="padding: 4px 4px 4px 4px;"
      >
        Currency
      </div>
      <div
        class="ant-col ant-col-4"
        style="padding: 4px 4px 4px 4px;"
      >
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-select ant-select-single ant-select-show-arrow"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                        aria-autocomplete="list"
                        aria-controls="rc_select_TEST_OR_SSR_list"
                        aria-haspopup="listbox"
                        aria-owns="rc_select_TEST_OR_SSR_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="rc_select_TEST_OR_SSR"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-placeholder"
                    >
                      Select
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row"
      style="margin: -4px -4px 4px -4px;"
    >
      <div
        class="ant-col ant-col-4 col1-text"
        style="padding: 4px 4px 4px 4px;"
      >
        LCN
      </div>
      <div
        class="ant-col ant-col-6"
        style="padding: 4px 4px 4px 4px;"
      >
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <button
                  aria-checked="false"
                  class="ant-switch"
                  data-testid="lcn-switch"
                  id="lcn"
                  role="switch"
                  type="button"
                >
                  <div
                    class="ant-switch-handle"
                  />
                  <span
                    class="ant-switch-inner"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row"
      style="margin: -4px -4px 4px -4px;"
    >
      <div
        class="ant-col ant-col-4"
        style="padding: 4px 4px 4px 4px;"
      />
      <div
        class="ant-col ant-col-6"
        style="padding: 4px 4px 4px 4px;"
      >
        <button
          class="ant-btn ant-btn-primary ant-btn-lg"
          data-testid="browse-supply-search-submit-btn"
          type="submit"
        >
          Search
        </button>
      </div>
    </div>
  </form>
</div>
`;
